src/lib.rs

```diff
@@ -1,6 +1,44 @@
 #![feature(get_mut_unchecked)]
 
-type Result<T> = std::result::Result<T, failure::Error>;
+use bitflags::_core::str::Utf8Error;
+use std::io;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Error in USB communication")]
+    RUSBError(#[from] rusb::Error),
+    #[error("IO Error")]
+    IOError(#[from] io::Error),
+    #[error("UTF8 Error")]
+    Utf8Error(#[from] Utf8Error),
+    #[error("{0}")]
+    GeneralError(String),
```

Comment:
More specific error types for various error cases woulDbe nice, but I guess this is fine for now :)